shared/middleware/helpers/shared_auth_helpers.rb

```diff
@@ -0,0 +1,22 @@
+require 'cdo/user_helpers'
+require 'json'
+
+#
+# Utility methods that help middleware access dashboard authentication and
+# permissions information.
+#
+# The use of these methods within shared and pegasus is deprecated.
+# These methods should be eliminated or moved to dashboard.
+#
+
+# @returns [Integer] the user_id associated with the current request
+def current_user_id
+  # @request is a Sinatra::Request < Rack::Request provided by Sinatra::Base
+  @request.nil? ? nil : @request.user_id
+end
+
+# @returns [Hash] the dashboard user row associated with the current request.
+def current_user
+  return nil if (id = current_user_id).nil?
+  @dashboard_user ||= DASHBOARD_DB[:users][id: id]
+end
```

Comment:
`current_user` is still used in pegasus here: https://github.com/code-dot-org/code-dot-org/blob/4ce3940e5801b30a277456530013bd6391504a92/pegasus/sites.v3/code.org/views/header.haml#L42

since we are just checking whether the user is signed in, it looks possible to change it so that it gets that information from a cookie instead, like we do here: https://github.com/code-dot-org/code-dot-org/blob/dc24d3bc39b678c92755249c440e5a31b4f184ce/pegasus/sites.v3/code.org/public/index.haml#L18-L21 but I am going to save that for a future PR.